pkg/engine/internal/workflow/task.go

```diff
@@ -25,6 +25,9 @@ type Task struct {
 	// Sinks defines which Streams physical nodes write to. Sinks are only
 	// defined for nodes in the Fragment which write data across task boundaries.
 	Sinks map[physical.Node][]*Stream
+
+	// TimeRange is a time tange of data that is gonna be processed by this task.
```

Comment:
tiny nit: ideally we avoid slang in comments 😛 

```suggestion
	// TimeRange is a time range of data that will be processed by this task.
```